packages/server/src/async/producers/feed-items/member-joined-to-community.js

```diff
@@ -0,0 +1,73 @@
+const assert = require('assert');
+const _ = require('lodash');
+
+const { ProducerFeedItems } = require('./producer-feed-items');
+const { CHANNEL_MEMBERSHIP_TYPE } = require('../../../models/enums/channel-membership-type');
+
+const { Producer } = require('../producer');
+
+class MemberJoinedToCommunity extends ProducerFeedItems {
+
+  /**
+   * @param {Models.ChannelMembership} channelMembership
+   */
+  constructor(channelMembership) {
+    super({});
+
+    assert(_.isObject(channelMembership), 'Channel Membership should be an object');
+    assert(!!_.get(channelMembership, 'agentId'), 'Channel Membership Agent Id is required');
+    assert(!!_.get(channelMembership, 'channelId'), 'Channel Membership Channel Id is required');
+    assert(!!_.get(channelMembership, 'type'), 'Channel Membership Type is required');
```

Comment:
This seems like a lot of boilerplate. Is there any nicer way of doing validations? Maybe as a future story?

Reply:
Mmm... I believe that it is the most elegant way of all existed ways: readable, fast to change, "easy come, easy go",  messy but compelling